src/controllers/queryRunner.ts

```diff
@@ -1069,90 +1072,93 @@ export default class QueryRunner {
         selection: ISlickRange[],
         batchId: number,
         resultId: number,
+        showThresholdWarning: boolean = true,
     ): Promise<void> {
-        const sendCancelSummaryEvent = () => {
+        /** Ask the user to proceed for large selections. */
+        const waitForUserToProceed = async (
+            requestId: string,
+            totalRows: number,
+        ): Promise<void> => {
+            const proceed = new Deferred<void>();
             this.fireSummaryChangedEvent(requestId, {
-                text: LocalizedConstants.QueryResult.summaryLoadingCanceled,
-                tooltip: LocalizedConstants.QueryResult.summaryLoadingCanceledTooltip,
+                command: {
+                    title: Constants.cmdHandleSummaryOperation,
+                    command: Constants.cmdHandleSummaryOperation,
+                    arguments: [this.uri],
+                },
+                continue: proceed,
+                text: `$(play-circle) ${LocalizedConstants.QueryResult.summaryFetchConfirmation(totalRows)}`,
```

Comment:
A couple random questions about the summary (for my own learning):
* Is the summary calculated in MSSQL for just the rows that have been loaded, or for the entire resultset (in STS, I guess)?
* If the first one, is the summary updated as results are streamed in or do we wait until we've loaded everything we will

Reply:
We calculate summaries for selected items using STS. The selection can include only the currently loaded rows, or the entire table (e.g., "Select All" or selecting a column by clicking on column header). Previously, we only performed calculations on loaded rows, which was incorrect.  Bug filed for it: **https://github.com/microsoft/vscode-mssql/issues/20187**

Now, we load the selected rows directly from STS before running calculations. While it would be possible to skip this step when only visible rows are selected, I chose not to optimize for that case. Since users typically select only a handful of rows manually, the extra roundtrip is inexpensive and not worth maintaining separate flows.